### PR TITLE
Remove validationMessages from formInput hoc and compose inputs with withMessages hoc

### PR DESCRIPTION
**Pull request by alcferreira**
affects: @crave/farmblocks-hoc-input

Now the validation messages are handled by the withMessages hoc from hoc-validation-messages package

BREAKING CHANGE:
Packages that use this hoc now must also be composed with the withMessages hoc from
hoc-validation-messages in order to render the error messages.

ISSUES CLOSED: #261

### CODE REVIEW

**Comment by alcferreira** [comment]
We should update the `hoc-validation-messages` with the latest version before merge this PR.
We need https://github.com/CraveFood/farmblocks/pull/259 first.

**Comment by alcferreira** [comment]
Good to review

File: packages/hoc-input/src/formInput.test.js [hunk 1]
```diff
@@ -10,16 +10,14 @@ describe("formInput", function() {
...
+  test("passing the same value on props should leave value state as it is", function() {
+    const value = "some value";
+    const component = shallow(<EnhancedInput value={value} />);
 
-    const expectedState = { ...state, validationMessages };
-    expect(component.state()).toEqual(expectedState);
+    component.setProps({ value });
+
+    expect(component.state().value).toEqual(value);
   });
```
[the remark this answers is not in the document]
Comment (alcferreira) [reply packages/hoc-input/src/formInput.test.js]: Good catch!

[the remark this answers is not in the document]
Comment (alcferreira) [reply packages/hoc-input/src/formInput.test.js]: Actually I don't see why it's incomplete...

- state.value is set when component mounts
- Passing the same value on `setProps` should not update state. It doesn't matter if state was set by a previous `setState` or `setProps`
- Passing a different value, should update state (which is covered on the next test)

Let me know if I'm missing anything

[the remark this answers is not in the document]
Comment (vNakamura) [reply packages/hoc-input/src/formInput.test.js]: The problem is: we're testing the state with the same value we are giving in props, so it won't break if the condition is not checking correctly and the `value` state IS been replaced by the `value` prop.

Change line 129 to `if(true) {` here:
https://github.com/CraveFood/farmblocks/blob/4689ce1b98cf992e9e0371bce3931f814709992e/packages/hoc-input/src/formInput.js#L128-L134
The test still passes.

If, in the future, we need to change the `componentWillReceiveProps` and by mistake we let the state be changed no matter what. This test won't warn us about that.

[the remark this answers is not in the document]
Comment (alcferreira) [reply packages/hoc-input/src/formInput.test.js]: > If, in the future, we need to change the componentWillReceiveProps and by mistake we let the state be changed no matter what. This test won't warn us about that.

- Mount component with value = `A`. State.value is set to `A`.
- call setState and pass value = `B`. State.value is set to `B`
- call setProps and pass value = `B`. State.value should not be updated and remain with value `A`

Same thing will happen in this scenario.

We can mock `setState` and assert if it has never been called.
I think this is the only way we can be sure that the state won't be updated

[the remark this answers is not in the document]
Comment (vNakamura) [reply packages/hoc-input/src/formInput.test.js]: Yes, mocking `setState` is the answer! 😃

File: packages/input-select/package.json [hunk 2]
```diff
@@ -33,7 +33,7 @@
     "@crave/farmblocks-dropdown": "^0.5.6",
     "@crave/farmblocks-hoc-disabled-tooltip": "^1.1.0",
     "@crave/farmblocks-hoc-input": "^1.2.4",
-    "@crave/farmblocks-hoc-validation-messages": "^0.2.0-alpha.a3a35fea",
+    "@crave/farmblocks-hoc-validation-messages": "^0.2.0",
```
Comment (fczuardi) [inline packages/input-select/package.json]: 👍

**Pull request merged**